core/src/readwriter.rs

```diff
@@ -39,6 +39,732 @@ impl ToSchema for &str {
+            options.insert("header".to_string(), header.to_string());
+        }
+
+        if let Some(delimiter) = self.delimiter {
+            options.insert("delimiter".to_string(), (delimiter as char).to_string());
+        }
+
+        if let Some(null_value) = &self.null_value {
+            options.insert("null_value".to_string(), null_value.to_string());
+        }
+
+        options
+    }
+}
+
+/// A struct that represents options for configuring JSON file parsing.
+///
+/// By default, this supports `JSON Lines` (newline-delimited JSON).
+/// For single-record-per-file JSON, set the `multi_line` option to `true`.
+///
+/// If the `schema` option is not specified, the input schema is inferred from the data.
+///
+/// # Options
+///
+/// - `schema`: An optional schema for the JSON data, either as a `StructType` or a DDL string.
+/// - `primitives_as_string`: Treat primitive types (e.g., integers, booleans) as strings.
+/// - `prefers_decimal`: Prefer parsing numbers as decimals rather than floating points.
+/// - `allow_comments`: Allow comments in JSON files (e.g., lines starting with `//` or `/* */`).
+/// - `allow_unquoted_field_names`: Allow field names without quotes.
+/// - `allow_single_quotes`: Allow the use of single quotes instead of double quotes for strings.
+/// - `allow_numeric_leading_zero`: Allow numbers to have leading zeros (e.g., `007`).
+/// - `allow_backslash_escaping_any_character`: Allow backslashes to escape any character.
+/// - `mode`: The parsing mode (e.g., `PERMISSIVE`, `DROPMALFORMED`, `FAILFAST`).
+/// - `column_name_of_corrupt_record`: Name of the column where corrupted records are placed.
+/// - `date_format`: Custom date format (e.g., `yyyy-MM-dd`).
+/// - `timestamp_format`: Custom timestamp format (e.g., `yyyy-MM-dd HH:mm:ss`).
+/// - `multi_line`: Read multiline JSON files (e.g., when a single JSON object spans multiple lines).
+/// - `allow_unquoted_control_chars`: Allow unquoted control characters in JSON (e.g., ASCII control characters).
+/// - `line_sep`: Custom line separator (default is `\n`).
+/// - `sampling_ratio`: Fraction of the data used for schema inference (e.g., `0.1` for 10%).
+/// - `drop_field_if_all_null`: Drop fields that are `NULL` in all rows.
+/// - `encoding`: Character encoding (default is `UTF-8`).
+/// - `locale`: Locale for parsing dates and numbers (e.g., `en-US`).
+/// - `path_glob_filter`: A glob pattern to filter files by their path.
+/// - `recursive_file_lookup`: Enable recursive search for files in directories.
+/// - `allow_non_numeric_numbers`: Allow special non-numeric numbers (e.g., `NaN`, `Infinity`).
+///
+/// # Example
+/// ```
+/// let options = JsonOptions::new()
+///     .schema("name STRING, salary INT")
+///     .multi_line(true)
+///     .allow_comments(true)
+///     .encoding("UTF-8".to_string());
+///
+/// let df = spark.read().json("/path/to/json", options)?;
+/// ```
+#[derive(Debug)]
+pub struct JsonOptions {
```

Comment:
Could you also add in these options?

- `time_zone`
- `timestamp_ntz_format`
- `enable_datetime_parsing_fallback`
- `compression`
- `ignore_null_fields`

https://spark.apache.org/docs/latest/sql-data-sources-json.html

Reply:
Right away!